core/src/test/java/org/fao/geonet/kernel/security/ldap/LDAPUserDetailsContextMapperWithProfileSearchEnhancedTest.java

```diff
@@ -135,6 +133,29 @@
     @Autowired
     LDAPRoleConverterGroupNameParser ldapRoleConverterGroupNameParser;
 
+
+    //=------------------------------------------------------------------
+
+    @AfterClass
+    public static void afterClass() throws Exception {
+        shutdownLDAP();
+    }
+
+    @BeforeClass
+    public static void beforeClass() throws Exception {
+        setupLDAP();
```

Comment:
This change seems to be the cause of the test failure, by by moving this to beforeclass and after class (rather than before and after test) the apache directory is still running, and using its own eccache.

Apparently each JVM can only have one eccache, and both hibernate and apache directory server want to make use of the technology.

Reference:
- [CacheService](https://directory.apache.org/apacheds/gen-docs/2.0.0-M24/apidocs/org/apache/directory/server/core/api/CacheService.html)

Reply:
One "fix" is to ugprade to ApacheDS 2.0.0.AM26 which replaces ecache with Caffeine. 

> We also have changed the Cache system we were using (ehcache) to use the more efficient and lighter [Caffeine](https://github.com/ben-manes/caffeine/wiki).

Reply:
We are trying this out, but requires some changes to newer maven and so on ...

Reply:
Reverted to previous way of setting up / down the ldap server.
And run ` CacheManager.getInstance().shutdown()` in `@AfterClass` to fix the problem with multiple chache managers in shibboleth tests.